app/models/job_execution.rb

```diff
@@ -109,6 +110,7 @@ def execute!(dir)
       "export REVISION=#{@reference.shellescape}",
       "export TAG=#{(@job.tag || @job.commit).to_s.shellescape}",
       "export CACHE_DIR=#{artifact_cache_dir}",
+      "export DEPLOY_GROUPS='#{@stage.deploy_groups.map(&:name).join(' ') if @stage}'",
```

Comment:
setting an empty set would be a little bit of clutter for non-deploy-grouped deploys
->

```Ruby
xxx = []
if @stage && deploy_groups = @stage.deploy_groups.map(&:name).join(' ').presence
  xxx << "export DEPLOY_GROUPS='#{deploy_groups}'"
end
```